src/app/core/services/user_service.py

```diff
@@ -275,28 +282,93 @@ def user_exists(user_id: int, session: ScopedSession) -> bool:
 	"""
 	return session.query(exists(User).where(User.user_id == user_id)).scalar()
 
+def email_exists(email: str, session: ScopedSession) -> bool:
+	"""Checks if a user with given email exists
+
+	Args:
+			email (str): email to check for
+			session (ScopedSession): SQLAlchemy database session
+
+	Returns:
+		bool: Represents whether user exists
+	"""
+	return session.query(exists(User).where(User.email == email)).scalar()
+
 def start_user_reset_password(email: str, session: ScopedSession):
 	"""Sends user email with authentication code for password reset
 
 	Args:
-			user_id (int): User ID to check for
+			email (str): email to check for
 			session (ScopedSession): SQLAlchemy database session
 
 	"""
-	#TODO: we dont have the user id, user has to enter their email to get the code
-	# if user_exists(user_id, session):
-		# user = _get_user_email(user_id, session)
+	code = verification.verify_code()
+	user =  _get_user_id(email, session)
+	if email_exists(email, session):
+		emailer.send_email(code, "Flourish Password Reset Code", email)
+		_set_password_reset_code(user, code, session)
+	else:
+		logging.error(f'User does not exist')
+		raise NotFoundError(f'Could not find user with email: {email}')
 
-		#TODO: replace the messege with the authenticator code
-		# emailer.send_email("This is a place holder", "Flourish Authentication Code", email)
-	# else:
-	# 	logging.error(f'User does not exist')
-	# 	raise NotFoundError(f'Could not find user with ID: {user_id}')
+def start_verify_password_reset_code(email: str, code: str, session: ScopedSession):
+	"""Verify if user-entered code is same as the code in the database
 
-	emailer.send_email("This is a place holder", "Flourish Authentication Code", email)
+	Args:
+			user_id (int): User ID
+			code (int): Verification/Password Reset Code
+			session (ScopedSession): Database
+
+	Raises:
+			e: _description_
+			NotFoundError: _description_
+
+	Returns:
+			_type_: _description_
+	"""
 
+	user_id = _get_user_id(email, session)
+	exists = verify_password_reset_code(user_id, code, session)
 
+	if exists == True:
+		return user_id
+	else:
+		return 0 
 
+def verify_verification_code(email: str, code: str, session: ScopedSession):
+	"""Verify if user-entered code is same as the code in the database
+
+	Args:
+			user_id (int): User ID
+			code (int): Verification/Password Reset Code
+			session (ScopedSession): Database
+
+	Raises:
+			e: _description_
+			NotFoundError: _description_
+
+	Returns:
+			_type_: _description_
+	"""
+	query = select(User).where(User.email == email)
+
+	try:
+		result: User = session.execute(query).scalar_one()
+	except exc.DatabaseError as e:
+		logging.error('Failed to get user password reset code')
+		logging.exception(e)
+		raise e
+	except exc.NoResultFound as e:
+		logging.error('Failed to find user')
+		logging.exception(e)
+		raise NotFoundError(f'Could not find user with the email: {email}')
```

Comment:
Switch exceptions, I think database error will catch everything before it reaches no result found